### DISCUSSION TITLE
Document implementation details for Doxygen (e.g. in c++ files)

### DISCUSSION THREAD

**Discussion by nmellado** [STORM-IRIT/Radium-Engine#773] [Ideas]
In PR #764 , @MathiasPaulin propose to introduce a variable `IMPLEMENTATION_DETAILS` to hide/show comments in the documentation generated by doxygen. The idea is to have `IMPLEMENTATION_DETAILS` enabled by default when compiling the documentation locally, but to disable it in the deployed documentation to ease reading for end-users.

What do you think ?

**Comment by MathiasPaulin** [comment]
Well, to be more precise, we can add in a next PR a cmake option (false by defaut so that the user must ask explicitely for doc details) to enable this so that so that local developer documentation include some implementation details of Radium internal. This demonstrates the way one can document such implementation details.
And, obviously, I agree ^^ with this.
